### PR TITLE
Create committee event

### PR DESCRIPTION
**Pull request by hhamon**

### CODE REVIEW

File: app/config/services/committee.xml [hunk 1]
```diff
@@ -13,13 +13,29 @@
...
         </service>
 
+
+        <service id="app.committee.committee_event_handler" class="AppBundle\Committee\Event\CommitteeEventCommandHandler">
+            <argument type="service" id="event_dispatcher"/>
+            <argument type="service" id="app.committee_event_factory"/>
+            <argument type="service" id="doctrine.orm.entity_manager"/>
+        </service>
+
+        <service id="app.committee.event_message_notifier" class="AppBundle\Committee\Event\CommitteeEventMessageNotifier">
```
Comment (HeahDude) [inline app/config/services/committee.xml]: This does not belong to the scope of this PR.

File: src/AppBundle/Collection/CommitteeMembershipCollection.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+namespace AppBundle\Collection;
+
+use AppBundle\Entity\CommitteeMembership;
+use Doctrine\Common\Collections\ArrayCollection;
+
+class CommitteeMembershipCollection extends ArrayCollection
+{
+    public function getAdherentUuids()
+    {
+        return array_map(
```
Comment (HeahDude) [inline src/AppBundle/Collection/CommitteeMembershipCollection.php]: `return $this->map(function (...)`.

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    ) {
+        $this->uuid = $uuid ?: Uuid::uuid4();
+        $this->committee = $committee;
+        $this->author = $author;
+        $this->address = $address ?: new Address();
+        $this->beginAt = $beginAt ?: new \DateTime(date('Y-m-d 00:00:00'));
+        $this->finishAt = $finishAt ?: new \DateTime(date('Y-m-d 23:59:59'));
+    }
+
+    public function getName()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `getName(): ?string`

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name)
+    {
+        $this->name = $name;
+    }
+
+    public function getCategory()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `: ?string`

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    {
+        return $this->category;
+    }
+
+    public function setCategory(string $category)
+    {
+        $this->category = $category;
+    }
+
+    public function getDescription()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `: ?string`

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+    {
+        return $this->description;
+    }
+
+    public function setDescription(string $description)
+    {
+        $this->description = $description;
+    }
+
+    public function getCapacity()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `: ?int`

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+    public function setCapacity($capacity)
+    {
+        if (null !== $capacity) {
+            $capacity = (string) $capacity;
+        }
+
+        $this->capacity = $capacity;
+    }
+
+    public function getBeginAt()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `: ?\DateTimeInterface`

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+    {
+        return $this->beginAt;
+    }
+
+    public function setBeginAt(\DateTime $beginAt)
+    {
+        $this->beginAt = $beginAt;
+    }
+
+    public function getFinishAt()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `: ?\DateTimeInterface`

File: src/AppBundle/Committee/Event/CommitteeEventCommand.php [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+    {
+        $this->finishAt = $finishAt;
+    }
+
+    public function setAddress(Address $address)
+    {
+        $this->address = $address;
+    }
+
+    public function getAddress()
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventCommand.php]: `: ?Address`

File: src/AppBundle/Committee/Event/CommitteeEventMessageNotifier.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+use AppBundle\Committee\CommitteeManager;
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Committee;
+use AppBundle\Entity\CommitteeEvent;
+use AppBundle\Mailjet\MailjetService;
+use AppBundle\Mailjet\Message\CommitteeEventNotificationMessage;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+class CommitteeEventMessageNotifier implements EventSubscriberInterface
```
Comment (HeahDude) [inline src/AppBundle/Committee/Event/CommitteeEventMessageNotifier.php]: This is for the next issue.

File: app/config/services/committee.xml (hunk 1 above)
Comment (hhamon) [inline app/config/services/committee.xml]: Yes it does. This is a notifier to send the Mailjet message.

File: src/AppBundle/Collection/CommitteeMembershipCollection.php (hunk 2 above)
Comment (hhamon) [inline src/AppBundle/Collection/CommitteeMembershipCollection.php]: Oh nice!  Good catch!

File: src/AppBundle/Form/CommitteeEventCommandType.php [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+                'required' => false,
+            ])
+        ;
+    }
+
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'data_class' => CommitteeEventCommand::class,
+            'years' => array_combine(range(2016, 2022), range(2016, 2022)),
```
Comment (HeahDude) [inline src/AppBundle/Form/CommitteeEventCommandType.php]: This should be dynamic:
```php
$years = range(date('y'), date('y') + 5);
// ...
['years' => array_combine($years, $years)]
```

Comment (hhamon) [inline src/AppBundle/Form/CommitteeEventCommandType.php]: Fixed.

**Comment by HeahDude** [comment]
Thank you @hhamon.

**Pull request merged**